frameworks/supervised/exec.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import os
+import tempfile as tmp
+import pandas as pd
+import numpy as np
+
+import sklearn
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+
+from frameworks.shared.callee import call_run, result, Timer
+
+log = logging.getLogger(os.path.basename(__file__))
+
+import supervised
+from supervised.automl import AutoML
+import shutil
+
+def run(dataset, config):
+    log.info("\n**** mljar-supervised ****\n")
+
+    is_classification = config.type == "classification"
+
+    X_train, X_test = dataset.train.X, dataset.test.X
+    y_train, y_test = dataset.train.y, dataset.test.y
+
+    X_train = pd.DataFrame(X_train, columns=[f"f{i}" for i in range(X_train.shape[1])])
+    X_test = pd.DataFrame(X_test, columns=[f"f{i}" for i in range(X_test.shape[1])])
+
+    y_train = y_train.flatten()
+    y_test = y_test.flatten()
+
+
+    # cast columns to have float types
+    for col in X_train.columns:
+        try: 
+            x1 = X_train[col].astype(float)
+            X_train[col] = x1
+            x2 = X_test[col].astype(float)
+            X_test[col] = x2
+        except Exception as e:
+            pass
+
+    print(
+        "We completely ignore the advice to optimize towards metric: {}.".format(
+            config.metric
+        )
+    )
+    
+    ml_task = None  # the AutoML will guess about ML task
+    # however, in case of multiclass classification, we will set the ml_task manually.
+    # AutoML assumes that multiclass calssification is when there is from 2 to 20 unique values.
+    # In this benchmark, there can be casses when it is much more classes than 20. That's why we set manually.
+    if is_classification and len(np.unique(y_train)) > 2:
+        ml_task = "multiclass_classification"
+
+    algorithms = ["Baseline", "Decision Tree", "Random Forest", "Xgboost", "LightGBM"]
+    if config.max_runtime_seconds <= 60:
+        algorithms = ["Baseline"]
```

Comment:
what happens if you don't specify the `algorithms` param?
One of the requirement for the integration in this app is that frameworks should run with their defaults (outside time, memory constraints, seed and other params that don't influence the training, like your `results_path`).

In your case, it would be preferable to call AutoML as follow:
```
   training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}

    automl = AutoML(
        results_path=results_path,
        ml_task=ml_task,
        total_time_limit=config.max_runtime_seconds,
        seed=config.seed,
        **training_params
    )
```

and let the framework use the default algorithms: user can always then create a custom definition if he wants to specify those params (see https://github.com/openml/automlbenchmark/blob/master/docs/HOWTO.md#run-a-framework-with-different-hyper-parameters)

Reply:
I just added our requirements, as I realize it was missing: https://github.com/openml/automlbenchmark/blob/master/docs/HOWTO.md#add-a-default-framework

Reply:
As suggested, I have used `training_params`.

Reply:
@sebhrusen I have one question. By default the `mljar-supervised` is producing explanations for models. The computation of explanation is very time-consuming. For normal usage,  explanations are needed. For benchmark reasons they should be switched off.

Should I switch-off explanations in `exec.py` or add param in `frameworks.yaml` ? I'd like to keep the switched-on in the `mljar-supervised` package.

Reply:
see https://github.com/openml/automlbenchmark/pull/105#discussion_r423230748